packages/lesswrong/components/ultraFeed/ultraFeedSettingsTypes.ts

```diff
@@ -196,42 +210,38 @@ export const getPostBreakpointLevel = (breakpoint: number | null | undefined): T
 };
 
 export interface SettingsFormState {
-  sourceWeights: Record<FeedItemSourceType, number | ''>;
-  postLevel0: TruncationLevel;
-  postLevel1: TruncationLevel;
-  postLevel2: TruncationLevel;
-  commentLevel0: TruncationLevel;
-  commentLevel1: TruncationLevel;
-  commentLevel2: TruncationLevel;
   incognitoMode: boolean;
-  quickTakeBoost: number;
-  // Advanced truncation settings
-  lineClampNumberOfLines: number | '';
-  postBreakpoints: (number | null | '')[];
-  commentBreakpoints: (number | null | '')[];
-  // Misc settings
-  ultraFeedSeenPenalty: number | '';
-  postTitlesAreModals: boolean;
+  sourceWeights: SourceWeightFormState;
+  displaySetting: DisplaySettingsFormState;
+  commentScoring: CommentScoringFormState;
+  threadInterestModel: ThreadInterestModelFormState;
 }
 
 export interface SettingsFormErrors {
-  sourceWeights: Record<FeedItemSourceType, boolean>;
-  postLevel0?: boolean;
-  postLevel1?: boolean;
-  postLevel2?: boolean;
-  commentLevel0?: boolean;
-  commentLevel1?: boolean;
-  commentLevel2?: boolean;
-  quickTakeBoost?: boolean;
-  // Advanced truncation errors
+  sourceWeights?: Record<FeedItemSourceType, boolean>;
   lineClampNumberOfLines?: boolean;
   postBreakpoints?: boolean;
   commentBreakpoints?: boolean;
-  // Misc errors
-  ultraFeedSeenPenalty?: boolean;
+  commentScoring?: ZodFormattedError<CommentScoringFormState, string> | null;
+  threadInterestModel?: ZodFormattedError<ThreadInterestModelFormState, string> | null;
 }
 
-export type TruncationGridFields = 'postLevel0' | 'postLevel1' | 'postLevel2' | 'commentLevel0' | 'commentLevel1' | 'commentLevel2';
+export type WithEmptyString<T> = T extends number ? number | '' :
```

Comment:
Pretty sure this can be collapsed to `T extends number | string ? T | '' : T;`, no?